crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -1954,6 +1954,68 @@ impl Display for SuiTransactionEffects {
     }
 }
 
+/// The response from processing a dev inspect transaction
+#[derive(Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename = "DevInspectResults", rename_all = "camelCase")]
+pub struct DevInspectResults {
+    /// Summary of effects that likely would be generated if the transaction is actually run.
+    /// Note however, that not all dev-inspect transactions are actually usable as transactions so
+    /// it might not be possible actually generate these effects from a normal transaction.
+    effects: SuiTransactionEffects,
+    /// Execution results (including return values) from executing the transactions
+    /// Currently contains only return values from Move calls
+    results: Result<Vec<(usize, SuiExecutionResult)>, String>,
+}
+
+#[derive(Clone, Serialize, Deserialize, JsonSchema)]
+pub struct SuiExecutionResult {
```

Comment:
are `mutable_reference_outputs` objects, if so is there a way we can map the values to objects?

SuiTypeTag for the type of return values sgtm, thanks for the quick move!

Reply:
They could be anything, since this lets you call anything. 

For example:
```
fun bump(counter: &mut u64, amount: u64) { *counter = *counter + u64 }
```
So in this case, you would get `mutable_reference_outputs` with `(0, /* some bytes */, TypeTag::u64)`